proof-producer/bin/proof-producer/include/nil/proof-generator/prover.hpp

```diff
@@ -541,6 +545,25 @@ namespace nil {
                 return true;
             }
 
+            bool save_binary_assignment_table_to_file(const boost::filesystem::path& output_filename) {
+                BOOST_LOG_TRIVIAL(info) << "writing binary assignment table to " << output_filename;
+                
+                if (!assignment_table_.has_value() || !table_description_.has_value()) {
+                    BOOST_LOG_TRIVIAL(error) << "no assignment table is currently loaded into the Prover";
+                    return false;
+                }
+
+                std::ofstream out(output_filename.string(), std::ios::binary | std::ios::out);
+                if (!out.is_open()) {
+                    BOOST_LOG_TRIVIAL(error) << "failed to open file " << output_filename;
+                }
+
+                assignment_table_writer<Endianness, BlueprintField> writer(out);
+                writer.write_binary_assignment( assignment_table_.value(), table_description_.value());
```

Comment:
May be better `static` method:
```
assignment_table_writer<Endianness, BlueprintField>::write_binary_assignment_table
```

Reply:
In that case we will need to pass ostream as an argument to each sub-function. It is not a big deal but I am not sure if it worths.

Reply:
I guess, no make sense introduce class for keep only reference to the stream. Common solution - 1. don't use class at all,  2. class with pure static methods.
Main concern for me - have to do 2 steps: create instance, call method, instead one call.
We have no any cases where need create instance once and call write method many times

Reply:
Ok, will rewrite with static class methods

Reply:
done